scormxblock/scormxblock.py

```diff
@@ -506,6 +510,26 @@ def _publish_progress(self, completion):
         """
         self.runtime.publish(self, 'completion', {'completion': completion})
 
+    def calc_progress_measure(self, scorm_data):
+        """
+        Returns the averaged progress_measure of all scos in the current scorm content
+        :return: progress_measure if found, else 0
+        """
+        progress_sum = 0
+        scos = scorm_data.get('scos', [])
+        for sco in scos.values():
+            sco_data = sco.get('data', {})
+            try:
+                progress_sum = progress_sum + float(sco_data.get('cmi.progress_measure', '0.0'))
+            except (ValueError, AttributeError):
+                pass
+        try:
```

Comment:
we can replace this try except with 
progress_measure = progress_sum / len(scos) if progress_sum else 0

Reply:
The except block catches ZeroDivisionError, given if condition can not be substituted in its place to perform same functionality. Please observe.